### PR TITLE
[Upgrade Ruff] Fix all errors raised from ruff

### LINKED ISSUE #8877: Pre-commit auto-updates break ruff
### What would you like to share?

In these failing tests, algorithms that have not been touched are broken due to the latest pre-commit update (https://github.com/TheAlgorithms/Python/commit/5aefc00f0f1c692ce772ddbc616d7cd91233236b)

https://github.com/TheAlgorithms/Python/actions/runs/5629561962/job/15254651560?pr=8875

Perhaps the reason of this error is that when the pre-commit pull request is made, the pre-commit file is changed, however the tests run from the old .pre-commit-config, not the new, changed one. This can be shown in the pr #8878, where the tests supposedly fail, when reverting the commit

### PR DESCRIPTION
**Pull request by CaedenPH**
### Describe your change:
Fixes all errors shown in the failing tests here

https://github.com/TheAlgorithms/Python/actions/runs/5629561962/job/15254651560?pr=8875

Fixes #8876 
Fixes #8877 
Fixes #8878 
Fixes #8880 

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by CaedenPH** [comment]
@cclauss Instead of reverting, this fixes all the new ruff rules

File: cellular_automata/game_of_life.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def __judge_point(pt: bool, neighbours: list[list[bool]]) -> bool:
     if pt:
         if alive < 2:
             state = False
-        elif alive == 2 or alive == 3:
+        elif alive in (2, 3):
```
Comment (cclauss) [inline cellular_automata/game_of_life.py:101]: As discussed in `ruff rule PLR1714`
> Use a `set` if the elements are washable.

```suggestion
        elif alive in {2, 3}:
```
Repeat below.

File: web_programming/convert_number_to_words.py [hunk 2]
```diff
@@ -90,9 +90,7 @@ def convert(number: int) -> str:
                 else:
                     addition = ""
                     if counter in placevalue:
-                        if current == 0 and ((temp_num % 100) // 10) == 0:
-                            addition = ""
-                        else:
+                        if current != 0 and ((temp_num % 100) // 10) != 0:
```
Comment (cclauss) [inline web_programming/convert_number_to_words.py:93]: I do not think this is exactly the same thing.

**Review by cclauss — COMMENTED** [review]
Please also upgrade the ruff version in `.pre-commit-config.yaml` or else do `pre-commit autoupgrade`.

**Comment by CaedenPH** [comment]
@cclauss I have fixed the changes you requested

**Review by cclauss — APPROVED** [review]
Nice!!

**Pull request merged**